tests/src/Interop/ResolveEvent/ResolveEventTests.cs

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+using System.IO;
+using System.Runtime.Loader;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+public class ALC : AssemblyLoadContext
+{
+    protected override Assembly Load(AssemblyName assemblyName)
+    {
+        return Assembly.Load(assemblyName);
+    }
+}
+
+public class ResolveEventTests
+{
+    static int HandlerTracker = 1;
+
+    public static int Main()
+    {
+        bool success = false;
+
+        // Events on the Default Load Context
+
+        try
+        {
+            AssemblyLoadContext.Default.ResolvingUnmanagedDll += HandlerFail;
+            NativeSum(10, 10);
+        }
+        catch (DllNotFoundException e)
+        {
+            success = HandlerTracker == 0;
+        }
+
+        try
+        {
+            AssemblyLoadContext.Default.ResolvingUnmanagedDll += HandlerPass;
+            success &= NativeSum(10, 10) == 20;
+            success &= HandlerTracker == 0;
+        }
+        catch (Exception e)
+        {
+            success = false;
+        }
+
+        // Events on a Custom Load Context
+
+        try
+        {
+            string currentDir = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+            string testAsmDir = Path.Combine(currentDir, "..", "TestAsm", "TestAsm");
+
+            ALC alc = new ALC();
+            alc.ResolvingUnmanagedDll += HandlerPass;
+
+            var assembly = alc.LoadFromAssemblyPath(Path.Combine(testAsmDir, "TestAsm.dll"));
+            var type = assembly.GetType("TestAsm");
+            var method = type.GetMethod("Sum");
+            success &= (int)method.Invoke(null, new object[] { 10, 10 }) == 20;
+        }
+        catch (Exception e)
+        {
+            success = false;
+        }
+
+        return (success) ? 100 : -100;
```

Comment:
Please follow the guidance in the Interop test `readme.md` for what main should look like.

Reply:
Can you please elaborate on what change is required? For `Main()` I followed the pattern in [test-configuration](https://github.com/dotnet/coreclr/blob/master/Documentation/building/test-configuration.md) pointed to from the [Readme](https://github.com/dotnet/coreclr/blob/master/tests/src/Interop/ReadMe.md)

Reply:
Sorry. Yep, the test-configuration. The logging of the exception, etc.